stock_picking_auto_create_lot/tests/test_stock_picking_auto_create_lot.py

```diff
@@ -144,3 +144,22 @@ def test_multi_auto_create_lot(self):
             [("product_id", "=", self.product_serial.id)]
         )
         self.assertEqual(len(lot), 6)
+
+    def test_immediate_validate_tracked_move_with_auto_create_lot(self):
+        picking_form = Form(self.env["stock.picking"])
+        picking_form.picking_type_id = self.picking_type_in
+        with picking_form.move_ids_without_package.new() as move:
+            move.product_id = self.product_serial
+            move.product_uom_qty = 4
+        receipt = picking_form.save()
+        receipt.action_confirm()
+
```

Comment:
I'd consider to do something like the following, upon adjusting `_create_move()` to have the picking as an arg.
```suggestion
        picking_1 = self.picking
        self._create_move(product=self.product_serial, qty=4.0, picking=picking_1)
        picking_1.action_assign()
```